sentry-types/src/protocol/attachment.rs

```diff
@@ -0,0 +1,80 @@
+use std::fmt;
+
+/// The different types an attachment can have
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub enum AttachmentType {
+    /// A standard attachment without special meaning. (default)
+    Attachment,
+    /// A minidump file that creates an error event and is symbolicated. The
+    /// file should start with the `MDMP` magic bytes.
+    Minidump,
+    /// An Apple crash report file that creates an error event and is symbolicated.
+    AppleCrashReport,
+    /// An XML file containing UE4 crash meta data. During event ingestion,
+    /// event contexts and extra fields are extracted from this file.
+    UnrealContext,
+    /// A plain-text log file obtained from UE4 crashes. During event ingestion,
+    /// the last logs are extracted into event breadcrumbs.
+    UnrealLogs,
+}
+
+impl Default for AttachmentType {
+    fn default() -> Self {
+        Self::Attachment
+    }
+}
+
+impl AttachmentType {
+    /// Gets the string value Sentry expects for the attachment type
+    pub fn as_str(self) -> &'static str {
+        match self {
+            Self::Attachment => "event.attachment",
+            Self::Minidump => "event.minidump",
+            Self::AppleCrashReport => "event.applecrashreport",
+            Self::UnrealContext => "unreal.context",
+            Self::UnrealLogs => "unreal.logs",
+        }
+    }
+}
+
+#[derive(PartialEq)]
+/// Represents an attachment item
+pub struct Attachment {
+    /// The actual attachment data
+    pub buffer: std::borrow::Cow<'static, [u8]>,
```

Comment:
I’m a bit unsure about usage of `Cow` here. All the other envelope items have `Cow`s inside, and expose that lifetime parameter.

Either way, I will look at the usage of Cow vs String in other places.

Reply:
I think a `Vec` would be the best idea here. I doubt you will have the case where you have a `'static` attachment buffer. Down the road, the user-facing attachment API will allow you to pass an owned buffer, or lazily load from disk at capture time, as is done in native today.

Reply:
Cool, done. :slightly_smiling_face: